pinot-core/src/main/java/org/apache/pinot/core/periodictask/PeriodicTaskScheduler.java

```diff
@@ -108,4 +108,48 @@ public synchronized void stop() {
       _tasksWithValidInterval.parallelStream().forEach(PeriodicTask::stop);
     }
   }
+
+  /** @return true if task with given name exists; otherwise, false. */
+  public boolean hasTask(String periodicTaskName) {
+    for (PeriodicTask task : _tasksWithValidInterval) {
+      if (task.getTaskName().equals(periodicTaskName)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  private PeriodicTask getPeriodicTask(String periodicTaskName) {
+    for (PeriodicTask task : _tasksWithValidInterval) {
+      if (task.getTaskName().equals(periodicTaskName)) {
+        return task;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Execute specified {@link PeriodicTask} immediately. If the task is already running, wait for the running task
+   * to finish before executing the task again.
+   */
+  public void execute(String periodicTaskName, String tableName) {
+    PeriodicTask periodicTask = getPeriodicTask(periodicTaskName);
+    LOGGER.info("Immediately executing periodic task {}", periodicTaskName);
+    if (periodicTask == null) {
+      throw new IllegalArgumentException("Unknown Periodic Task " + periodicTaskName);
+    }
+
+    _executorService.schedule(() -> {
+      try {
+        // To prevent thread conflict, this call will block if the same task is already running (see
```

Comment:
TBD whether we should queue and re-run the task immediately after one run. I would suggest don't bother. If it is running, just return with an INFO message saying so.

I am eager to hear the requirements here and what others have to say.